src/renderers/WebGLRenderer.js

```diff
@@ -1661,7 +1662,7 @@ function WebGLRenderer( parameters ) {
 
 				initMaterial( material, scene, object );
 
-			} else if ( materialProperties.outputEncoding !== _this.outputEncoding ) {
+			} else if ( materialProperties.outputEncoding !== _currentEncoding ) {
```

Comment:
Where is `_currentEncoding` updated when the user changes `WebGLRenderer.outputEncoding`?

Reply:
Oh, good point.

Reply:
Tried to update  `_currentEncoding` outside of `setProgram`, so that we didn't have to re-check for every rendered object, but `renderObjectDirect` and `renderObjectImmediate` can be called directly, so I'm afraid the best solution is putting it inside `setProgram` 😞

Reply:
If we decide to change API to use `setOutputEncoding()`, we can move the check outside.

Reply:
If we go for that `setOutputEncoding()` API change, we can prevent having to re-check everytime we render a object.

Reply:
Don't worry, this check if okay 👍 .